File: android/app/src/main/kotlin/com/zaplivre/voip/VoipEventHandler.kt
package com.zaplivre.voip

import android.content.Context
import android.util.Log
import uniffi.zaplivre.FfiVoipEventCallback

/**
 * VoipEventHandler - Receives VoIP control events from core and applies them on Android.
 */
class VoipEventHandler(context: Context) : FfiVoipEventCallback {
    private val audioManager = CallAudioManager(context.applicationContext)

    override fun onMuteChanged(callId: String, isMuted: Boolean) {
        audioManager.setMuted(isMuted)
        Log.i("VoipEventHandler", "Mute changed for $callId: $isMuted")
    }

    override fun onSpeakerphoneChanged(callId: String, enabled: Boolean) {
        audioManager.setSpeakerphone(enabled)
        Log.i("VoipEventHandler", "Speakerphone changed for $callId: $enabled")
    }

    override fun onCameraSwitchRequested(callId: String) {
        Log.i("VoipEventHandler", "Camera switch requested for $callId")
        // UI layer should handle actual camera switching.
    }
}

File: ios/ZapLivre/ZapLivre/VoIP/VoipEventHandler.swift
//
//  VoipEventHandler.swift
//  ZapLivre
//
//  Created by ZapLivre Team
//

import Foundation

/// Bridges VoIP control events from core to the app CallManager.
final class VoipEventHandler: NSObject, FfiVoipEventCallback {
    private weak var callManager: CallManager?

    init(callManager: CallManager) {
        self.callManager = callManager
    }

    func onMuteChanged(callId: String, isMuted: Bool) {
        DispatchQueue.main.async { [weak self] in
            self?.callManager?.handleMuteChanged(isMuted)
        }
    }

    func onSpeakerphoneChanged(callId: String, enabled: Bool) {
        DispatchQueue.main.async { [weak self] in
            self?.callManager?.handleSpeakerChanged(enabled)
        }
    }

    func onCameraSwitchRequested(callId: String) {
        DispatchQueue.main.async { [weak self] in
            self?.callManager?.handleCameraSwitchRequested()
        }
    }
}
